src/containers/Application/ApplicationList.tsx

```diff
@@ -23,29 +50,49 @@ const ApplicationList: React.FC = () => {
           <List.Item key={`app_selected_parameter_${value}`} content={key + ' : ' + value} />
         ))}
       </List>
-      {applications && applications.length > 0 && (
+      {userRole && headers && applications && applications.length > 0 && (
         // TODO: Implement sortable columns -> Requires reducer
         <Table sortable stackable selectable>
           <Table.Header>
             <Table.Row>
-              {Object.entries(applications[0]).map(([key]) => (
-                <Table.HeaderCell key={`app_header_${key}`}>{key}</Table.HeaderCell>
+              {headers.map((headerName) => (
+                <Table.HeaderCell key={`app_header_${headerName}`}>{headerName}</Table.HeaderCell>
               ))}
             </Table.Row>
           </Table.Header>
           <Table.Body>
             {applications.map((application, index) => (
               <Table.Row key={`${application.serial}`}>
-                {Object.entries(application).map(([key, value]) => {
-                  switch (key) {
-                    case 'name':
+                {headers.map((headerName) => {
+                  // TODO: Move this to render property to be defined for each ColumnDetails in mapColumnsByRole
+                  switch (headerName) {
```

Comment:
yes, that's right. I haven't thought yet how exactly this would be done yet - I'm planning to update #207 with the approach before I start the implementation. But briefly this would be replaced to just display the component returned by the current `header` which would expect the current application to be passed and return how this cell would be displayed.